sys/net/gnrc/network_layer/ndp/internal/gnrc_ndp_internal.c

```diff
@@ -179,11 +179,11 @@ void gnrc_ndp_internal_send_nbr_adv(kernel_pid_t iface, ipv6_addr_t *tgt,
 
         if (l2src_len > 0) {
             /* add target address link-layer address option */
-            pkt = gnrc_ndp_opt_tl2a_build(l2src, l2src_len, NULL);
+            pkt = gnrc_ndp_opt_tl2a_build(l2src, l2src_len, pkt);
 
             if (pkt == NULL) {
                 DEBUG("ndp internal: error allocating Target Link-layer address option.\n");
-                gnrc_pktbuf_release(pkt);
+                gnrc_pktbuf_release(ext_opts);
```

Comment:
Hm, then I would opt for a good documentation for now, stating that `ext_ops` can be modified after the call to `gnrc_ndp_internal_send_nbr_adv`